crates/experimentation_client/src/lib.rs

```
@@ -67,19 +67,21 @@ impl Client {
                     *start_date,
                     self.client_config.tenant.to_string(),
                 )
-                .await
-                .unwrap_or(HashMap::new());
-
-                let mut exp_store = self.experiments.write().await;
-                for (exp_id, experiment) in experiments.into_iter() {
-                    if experiment.status.active() {
-                        exp_store.insert(exp_id, experiment)
-                    } else {
-                        exp_store.remove(&exp_id)
-                    };
-                }
+                .await;
+                match experiments {
+                    Ok(experiments) => {
+                        let mut exp_store = self.experiments.write().await;
+                        *exp_store = experiments;
+                        *start_date = Utc::now();
+                    }
+                    Err(e) => {
+                        log::error!(
+                            "Failed to fetch experiments from the server with error: {}",
+                            e
+                        );
+                    }
```

Just logging the error isn't sufficient here. Either fail loudly by panicking or provide a default. The loop may also need to be exited

---

why should we fail ?
if one iteration fails, it can just go on safely to the next iteration, the previous data would anyways be there, right ?